pkg/propertychecker/azure/checker.go

```diff
@@ -0,0 +1,171 @@
+	// The capacity is checked by ensuring the current allocatable capacity is greater than the requested capacity.
+	request := &computev1.GenerateAttributeBasedRecommendationsRequest{
+		SubscriptionId: subID,
+		Location:       location,
+		RegularPriorityProfile: &computev1.RegularPriorityProfile{
+			CapacityUnitType: computev1.CapacityUnitType_CAPACITY_UNIT_TYPE_VM_INSTANCE_COUNT,
+			TargetCapacity:   capacity, // CurrentAllocatableCapacity > RequestedCapacity
+		},
+		ResourceProperties: &computev1.ResourceProperties{
+			VmAttributes: &computev1.VMAttributes{
+				AllowedVmSizes: []string{sku},
+			},
+		},
+		RecommendationProperties: &computev1.RecommendationProperties{
+			RestrictionsFilter: computev1.RecommendationProperties_RESTRICTIONS_FILTER_QUOTA_AND_OFFER_RESTRICTIONS,
+		},
+	}
+
+	respObj, err := s.vmSizeRecommenderClient.GenerateAttributeBasedRecommendations(context.Background(), request)
+	if err != nil {
+		return false, fmt.Errorf("failed to generate VM size recommendations from Azure: %w", err)
+	}
+
+	// This check is a defense mechanism; vmSizeRecommenderClient should return a VM size recommendation
+	// if the SKU is available in the specified location and subscription.
+	available := false
+	for _, vm := range respObj.RecommendedVmSizes.RegularVmSizes {
+		if vm.Name == sku {
+			available = true
+			klog.V(2).Infof("SKU %s is available in cluster %s", sku, cluster.Name)
+			break
+		}
+	}
+
+	return available, nil
+}
+
+// ExtractCapacityRequirements extracts the capacity value from a PropertySelectorRequirement.
+// This function is specifically designed for Azure SKU capacity properties that follow the pattern:
+// "kubernetes.azure.com/vm-sizes/{sku}/capacity"
+// Returns the capacity as an uint32 if the requirement is valid;
+// or an error if the requirement is invalid.
+func ExtractCapacityRequirements(req placementv1beta1.PropertySelectorRequirement) (uint32, error) {
+	if req.Operator != placementv1beta1.PropertySelectorGreaterThan && req.Operator != placementv1beta1.PropertySelectorGreaterThanOrEqualTo {
+		return 0, fmt.Errorf("unsupported operator %q for SKU capacity property, only GreaterThan (Gt) and GreaterThanOrEqualTo (Ge) are supported", req.Operator)
+	}
+
+	// Validate that we have exactly one value.
+	if len(req.Values) != 1 {
+		return 0, fmt.Errorf("azure SKU capacity property must have exactly one value, got %d", len(req.Values))
+	}
+
+	capacity, err := validateCapacity(req.Values[0], req.Operator)
+	if err != nil {
+		return 0, fmt.Errorf("failed to validate capacity value %q: %w", req.Values[0], err)
+	}
+
+	// Safe conversion to uint32 - all validations passed
+	return capacity, nil
+}
+
+// validateCapacity checks if the provided capacity value is valid.
+// Returns the capacity as uint32 if valid, or a zero and an error if invalid.
+// If the operator is GreaterThanOrEqualTo, the returned capacity is adjusted by subtracting 1
+// to align with GreaterThan semantics.
```

Comment:
```suggestion
```
Better to put the context on the ExtractCapacityRequirements, instead of these internal func.

The comment is to tell the reader why we're doing this, not just what we're doing.